lib/bitex_bot/models/api_wrappers/kraken/kraken_order.rb

```diff
@@ -1,101 +1,114 @@
 require 'kraken_client'
 
+##
+# Wrapper for Kraken orders.
+#
 class KrakenOrder
   cattr_accessor :last_closed_order
   attr_accessor :id, :amount, :executed_amount, :price, :avg_price, :type, :datetime
 
-  # id: 'O5TDV2-WDYB2-6OGJRD'
-  # order_data: {
-  #     'refid': nil, 'userref': nil, 'status': 'open', 'opentm': 1440292821.4839, 'starttm': 0, 'expiretm': 0,
-  #     'descr': {
-  #       'pair': 'ETHEUR', 'type': 'buy', 'ordertype': 'limit', 'price': '1.19000', 'price2': '0', 'leverage': 'none',
-  #       'order': 'buy 1204.00000000 ETHEUR @ limit 1.19000'
-  #     },
-  #     'vol': '1204.00000000', 'vol_exec': '0.00000000', 'cost': '0.00000', 'fee': '0.00000', 'price': '0.00000',
-  #     'misc': '', 'oflags': 'fciq'
-  #   }
-  # }
-  def initialize(id, order_data)
-    self.id = id
-    self.amount = order_data[:vol].to_d
-    self.executed_amount = order_data[:vol_exec].to_d
-    self.price = order_data[:descr][:price].to_d
-    self.avg_price = order_data[:price].to_d
-    self.type = order_data[:descr][:type].to_sym
-    self.datetime = order_data[:opentm].to_i
-  end
-
-  def cancel!
-    KrakenApiWrapper.client.private.cancel_order(txid: id)
+  # rubocop:disable Metrics/AbcSize
+  def self.create!(type, price, quantity)
+    self.last_closed_order = closed.first.try(:id) || Time.now.to_i
+    find(order_info_by(type, price.truncate(1), quantity.trucante(8))['txid'].first)
   rescue KrakenClient::ErrorResponse => e
-    retry if e.message == 'EService:Unavailable'
-    raise
+    # Order could not be placed
+    if e.message == 'EService:Unavailable'
+      BitexBot::Robot.logger.debug('Captured EService:Unavailable error when placing order on Kraken. Retrying...')
+      retry
+    elsif e.message.start_with?('EGeneral:Invalid')
+      BitexBot::Robot.logger.debug("Captured #{e.message}: type: #{type}, price: #{price}, quantity: #{quantity}")
+      raise OrderArgumentError, e.message
+    elsif e.message != 'error'
+      raise
+    end
   end
+  # rubocop:enable Metrics/AbcSize
 
-  def ==(order)
-    if order.is_a?(self.class)
-      id == order.id
-    elsif order.is_a?(Array)
-      [type, price, amount] == order
-    end
+  def self.order_info_by(type, price, quantity)
+    KrakenApiWrapper.client.private.add_order(pair: 'XBTUSD', type: type, ordertype: 'limit', price: price, volume: quantity)
   end
 
   def self.find(id)
     new(*KrakenApiWrapper.client.private.query_orders(txid: id).first)
-  rescue KrakenClient::ErrorResponse => e
+  rescue KrakenClient::ErrorResponse
     retry
   end
 
-  def self.amount_and_quantity(order_id, transactions)
+  def self.amount_and_quantity(order_id)
```

Comment:
This will blow up, see https://github.com/bitex-la/bitex-bot/pull/16/files#diff-a958d92aeb89c26d068f4c7c17d948c5R18

Reply:
fixed!